File: Java/Searching/BinarySearch.java
package Searching;
/*Binary search, also known as half-interval search, logarithmic search,
or binary chop, is a search algorithm that finds the position of a 
target value within a sorted array. Binary search compares the target 
value to the middle element of the array.
Worst complexity: O(log n)
Average complexity: O(log n)
Best complexity: O(1)
Space complexity: O(1)
*/
public class BinarySearch {
    public static void main(String[] args) {
        int arr[] = {10, 18, 19, 23, 30, 89, 100};
        int target = 18;
        int start = 0;
        int end = arr.length - 1;
        int index = -1;
        while(start <= end) {
            //int mid = (end + start) / 2; -> might cause overflow
            int mid = start + (end - start) / 2;

            if(arr[mid] < target) {
                start = mid + 1;
            }
            else if(arr[mid] > target) {
                end = mid - 1;
            }
            else {
                System.out.println("Found");
                index = mid;
                break;
            }
        }

        if(start > end) {
            System.out.println("Not found");
        }

        System.out.println("Index: " + index);
    }
}

File: PHP/searching/binary_search.php

<?php
// Binary Search in PHP
function binarySearch(Array $arr, $item) {
    $low = 0;
    $high = count($arr) - 1;
	$got = false;

	while ($low <= $high) {
		$mid = $low + ($high - $low) / 2;

		if ($arr[$mid] === $item) { 
			$got = true;
			break;
		} else {
			if ($item < $arr[$mid]) { 
				$high = $mid - 1;
			} else { 
				$low = $mid + 1;
			}
		}
	}

	return $got;
}

$array = array(0, 1, 3, 5, 15, 16, 35, 48, 56, 73, 98 );

//for true
$search = 16;
echo binarySearch($array, $search) ? 'Found' : 'Not Found';
echo PHP_EOL;

echo '<br />';

//for false
$search = 6;
echo binarySearch($array, $search) ? 'Found' : 'Not Found';
echo PHP_EOL;

?>